### DISCUSSION TITLE
Feature Request: zsh/bash tsh tab auto completion

### DISCUSSION THREAD

**Discussion by luckypoint** [gravitational/teleport#40707] [Ideas]
It would be super helpful if tsh had tab auto completion, much like when use kubectl I can type just the beginning of pod name ("kubectl describe pod m...." hit tab here and it will fill the pod name ""kubectl describe pod my-pod-abc123").  It would be really nice to have the same functionality with "tsh ssh m..." which could auto complete to "tsh ssh my-server-aws-01" etc.

**Comment by stevenGravy** [comment]
Please take a look at https://goteleport.com/docs/reference/cli/#improve-the-cli-experience-enable-shell-completion. That should do it.

**Reply by zmb3** [reply]
Our shell completion currently only completes commands, not resources.

Completion for resources is tracked at #30175, but we will probably end up implementing our own completion system instead of using bash/zsh completion.

**Comment by luckypoint** [comment]
Thanks @zmb3 I am going to close this discussion as the linked one 30175 already covers my request.
